SmartTutor/ContentModel/LearningObjects/ChallengeModel/Challenge.cs

```diff
@@ -0,0 +1,21 @@
+﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
+using SmartTutor.ContentModel.LearningObjects.ChallengeModel.FulfillmentStrategy;
+using SmartTutor.ContentModel.LearningObjects.ChallengeModel.MetricRules;
+using System.Collections.Generic;
+
+namespace SmartTutor.ContentModel.LearningObjects.ChallengeModel
+{
+    public class Challenge : LearningObject
+    {
+        public string Url { get; internal set; }
+        public List<CaDETClass> ResolvedClasses { get; internal set; }
+        public List<MetricRangeRule> ClassMetricRules { get; internal set; }
```

Comment:
The MetricRangeRules are part of the basic challenge fulfillment strategy. For example, if we want to have an "AdequateNameChecker" as an implementation of the FulfillmentStrategy, we would host the key words we want to see in some name in this NameChecker, not in the Challenge.

This would also enable you to avoid passing the Challenge itself to the strategy, reducing the param. list and the feature envy code smell (since the strategy is extracting data from the Challenge currently).